web/assets/header-menu-toggle.js

```diff
@@ -0,0 +1,42 @@
+export default function toggleHeaderMenu() {
+    const headerContainer = document.querySelector(
+        '[data-module="app-header-container"]',
+    );
+
+    if (headerContainer) {
+        const menuButton = headerContainer.querySelector('[data-module="app-header-menu-toggle"]');
+        const menuItemList = headerContainer.querySelector('[data-id="app-header-menu-item-list"]')
+
+        if (menuButton && menuItemList) {
+            const tabletMQry = window.matchMedia('(max-width: 768px)');
+
+            if (tabletMQry.matches) {
+                menuItemList.hidden = true;
+            } else {
+                menuButton.hidden = true;
+            }
+
+            menuButton.addEventListener("click", () => {
+                if (menuItemList.hidden) {
+                    menuItemList.hidden = false;
+                    menuButton.ariaExpanded = "true";
+                } else {
+                    menuItemList.hidden = true;
+                    menuButton.ariaExpanded = "false";
+                }
+            });
+
+            tabletMQry.onchange = (e) => {
+                if (e.matches) { // screen is 0 - 768px
+                    menuItemList.hidden = true;
+                    menuButton.ariaExpanded = "false";
+                    menuButton.hidden = false;
+                } else { // screen is 768 -> bigger
+                    menuItemList.hidden = false;
+                    menuButton.ariaExpanded = "true";
+                    menuButton.hidden = true;
```

Comment:
This feels messy having media queries in the JS as well as the SCSS - it was necessary due to an override issue with the menu toggle button which appears on smaller screen widths: [GDS header component](https://design-system.service.gov.uk/components/header/with-service-name-and-navigation/) . I originally had a solution for the menu toggle button, using media queries in the SCSS file to show/hide it based on screen width, however the JS from the standard component `govuk-header__menu-button` was overriding my SCSS, to which I was then having to add `!important` tags to my SCSS which felt messier?